src/pages/Bridge/index.tsx

```diff
@@ -81,83 +159,268 @@ const BridgePage: React.FC = () => {
+  useEffect(() => {
+    const val =
+      !!getXvsBridgeStatusData &&
+      xvsAmountUsd.gt(getXvsBridgeStatusData.maxSingleTransactionLimitUsd);
+
+    setValue('isSingleTransactionLimitExceeded', val, { shouldValidate: true });
+  }, [getXvsBridgeStatusData, setValue, xvsAmountUsd]);
+
+  // check if the global daily limit of XVS transferred was reached
+  useEffect(() => {
+    const val = xvsAmountUsd.gt(remainingXvsDailyLimitUsd);
+
+    setValue('isDailyTransactionLimitExceeded', val, { shouldValidate: true });
+  }, [remainingXvsDailyLimitUsd, setValue, xvsAmountUsd]);
+
   const switchChains = useCallback(() => {
     const formValues = getValues();
 
     handleChainFieldChange({
-      fromChainId: formValues.toChainId,
-      toChainId: formValues.fromChainId,
+      newFromChainId: formValues.toChainId,
+      newToChainId: formValues.fromChainId,
     });
   }, [getValues, handleChainFieldChange]);
-
-  const onSubmit = (values: z.infer<typeof formSchema>) => {
-    // TODO: send transaction (see VEN-2248)
-    console.log(values);
+  const { mutateAsync: bridgeXvs } = useBridgeXvs();
+
+  const onSubmit = async () => {
+    if (accountAddress) {
+      try {
+        await bridgeXvs({
+          accountAddress,
+          amountMantissa,
+          destinationChainId: toChainId,
+          nativeCurrencyFeeMantissa: bridgeEstimatedFeeMantissa,
+        });
+      } catch (error) {
+        displayMutationError({ error });
+      }
+    }
   };
 
-  // TODO: wire up
-  const walletBalanceTokens = new BigNumber(10);
-  const limitTokens = walletBalanceTokens;
+  const errorLabel = useMemo(() => {
+    if (isSingleTransactionLimitExceeded) {
+      const xvsPriceUsd = getTokenUsdPriceData?.tokenPriceUsd || new BigNumber(0);
+      const maxSingleTransactionLimitTokens = maxSingleTransactionLimitUsd.dividedBy(xvsPriceUsd);
+      const readableAmountTokens = formatTokensToReadableValue({
+        value: maxSingleTransactionLimitTokens,
+        token: xvs,
+      });
+      const readableAmountUsd = formatCentsToReadableValue({
+        value: convertDollarsToCents(maxSingleTransactionLimitUsd),
+      });
+      return t('bridgePage.errors.singleTransactionLimitExceeded.message', {
+        readableAmountTokens,
+        readableAmountUsd,
+      });
+    }
+    if (isDailyTransactionLimitExceeded) {
+      const readableAmountTokens = formatTokensToReadableValue({
+        value: remainingXvsDailyLimitTokens,
+        token: xvs,
+      });
+      const readableAmountUsd = formatCentsToReadableValue({
+        value: convertDollarsToCents(remainingXvsDailyLimitUsd),
+      });
+      const date = fromUnixTime(dailyLimitResetTimestamp.toNumber());
+      date.setDate(date.getDate() + 1);
+      return t('bridgePage.errors.dailyTransactionLimitExceeded.message', {
+        readableAmountTokens,
+        readableAmountUsd,
+        date,
+      });
+    }
```

Comment:
Following my previous comment about using Zod's error handling, then you'd be able to get rid of this logic (or rather rewrite it into the validation schema) and get the error to display from the form state directly.